at_secondary/at_secondary_server/lib/src/verb/handler/scan_verb_handler.dart

```diff
@@ -60,12 +60,18 @@ class ScanVerbHandler extends AbstractVerbHandler {
       if (forAtSign != null &&
           atConnectionMetadata.isAuthenticated &&
           forAtSign != currentAtSign) {
-        response.data =
-            await _getExternalKeys(forAtSign, scanRegex, atConnection);
+        if (page == null) {
+          response.data =
```

Comment:
I don't think we need a if-else here. When we ask an external secondary for a scan with the page its response should be ready to be sent back without any work. We should be sending the page based syntax to it.